src/main/resources/lib/meta-synchronization/meta-synchronization.ts

```diff
@@ -0,0 +1,184 @@
+    'formDetailsTargets',
+    'feedbackToggle',
+    'chatbotToggle',
+    'hideFromProductlist',
+];
+
+type MetaData = { [key: string]: unknown };
+
+const checkIfMetaHasBeenEdited = (
+    content: DynamicPageContent,
+    previousContent: DynamicPageContent
+) => {
+    const isMetaChanged = metaDataToSync.some((key) => {
+        // Some meta data is stored as arrays or objects, and we need to compare them as JSON strings.
+        // If the keys or array order has changed, we might get a false positive here.
+        return JSON.stringify(content.data[key]) !== JSON.stringify(previousContent.data[key]);
+    });
+
+    return isMetaChanged;
+};
+
+const buildMetaDataObject = (content: DynamicPageContent): MetaData => {
+    const metaData = metaDataToSync.reduce((acc, key) => {
+        return content.data[key] ? { ...acc, [key]: content.data[key] } : acc;
+    }, {} as MetaData);
+
+    return metaData;
+};
+
+const syncToAllOtherLayers = (content: DynamicPageContent) => {
+    const { repoIdToLocaleMap } = getLayersData();
+
+    // Check previous version of content for change.
+    const versions = getNodeVersions({
+        nodeKey: content._id,
+        repoId: CONTENT_ROOT_REPO_ID,
+        branch: 'master',
+    });
+    const previousVersionId = versions[1]?.versionId;
+    const previousContentVersion =
+        previousVersionId &&
+        contentLib.get({
+            key: content._id,
+            versionId: previousVersionId,
+        });
+
+    const hasMetaBeenEdited = !!(
+        previousContentVersion && checkIfMetaHasBeenEdited(content, previousContentVersion)
+    );
+
+    if (!hasMetaBeenEdited) {
+        return;
+    }
+
+    const metaData = buildMetaDataObject(content);
+
+    Object.keys(repoIdToLocaleMap).forEach((repoId) => {
+        if (repoId === CONTENT_ROOT_REPO_ID) {
+            return;
+        }
+
+        // Get the draft and master repo and content in order to check
+        // for versionKey later.
+        const draftRepo = getRepoConnection({ repoId, branch: 'draft', asAdmin: true });
+        const masterRepo = getRepoConnection({ repoId, branch: 'master', asAdmin: true });
+        const draftContent = draftRepo.get<DynamicPageContent>({ key: content._id });
+        const masterContent = masterRepo.get<DynamicPageContent>({ key: content._id });
+
+        if (!draftContent) {
+            logger.error(
+                `Meta synchronization: Could not get content from ${repoId} layer with id ${content._id} when trying to copy meta data`
+            );
+            return;
+        }
+
+        if (!isContentLocalized(draftContent)) {
+            return;
+        }
+
+        const isPublished = masterContent && masterContent._versionKey === draftContent._versionKey;
+
+        try {
+            draftRepo.modify({
+                key: draftContent._id,
+                editor: (node) => {
+                    return { ...node, data: { ...node.data, ...metaData } };
+                },
```

Comment:
Dette vil skape problemer, ettersom feltene for id-referanser egentlig er verdier av Java-typen "Reference", selv om det ser ut som strings i Javascript/Nashorn. Se https://developer.enonic.com/docs/xp/stable/api/lib-value for litt forklaring av slike typer.

Må enten bruke Value API'et og sette riktige typer for det som ikke bare skal være plain strings, eller bruker content API modify, som håndterer dette automatisk ut fra schemaet for content-typen.